src/cpu/core_dyn_x86/decoder.h

```diff
@@ -2180,7 +2180,7 @@ static CacheBlock * CreateCacheBlock(CodePageHandler * codepage,PhysPt start,Bit
 	decode.block->page.start=decode.page.index;
 	codepage->AddCacheBlock(decode.block);
 
-	auto cache_addr = static_cast<void *>(const_cast<uint8_t *>(decode.block->cache.start));
+	auto cache_addr = static_cast<void *>(const_cast<uint8_t *>(decode.block->cache.start-1));
```

Comment:
Suggest spacing the ` - 1` here and below just to help visibility (with two casts in play, the -1 kind of gets lost in the activity :-)